### Two Tailed - One-Sample Z-Test

**1 - What is it?**

A. `Two-Tailed One-Sample Z-Test` is used to check whethher a sample mean is **significantly different (either higher or lower)** than the population mean, when the *population standard deviation known* and the *sample size is large (n>30)*.

**In other words:**
It checks if the sample mean devisates from the population mean *in either direction* not just greater or smaller, but any signficant difference.

**Example:**
Is my sample group's average score significantly different from the country's average, either higher or lower, or is the difference just due to random variation?

**2 - Real-Life Example: Two-Tailed One-Sample Z-Test**

`Scenario:` A Company claims that their delivery time averages **3 days**. A manager believes that the actual delivery time might be **different** (either faster or slower).

To verify this, we test whether the sample mean differs significantly from the claimed mean, in either direction, so this is a **tow-tailed test**.

**3 - Hypotheses**

- H0 (Null Hypothesis) : population mean = 3 -> The population mean delivery time is 3 days.
- H1 (Alternative Hypothesis) : population mean != 3 -> The delivery time is significantly different from 3 days (could be higher or lower)

This is a **two-tailed test** because we're testing for any *difference*, not just an *increase or decrease*.

**4 - Formula & Parameters**

Z = (sample mean - population mean) / (population std/sqrt(sample size))

Z = (x bar - μ) / (σ/sqrt(n))

**Parameters**
- Population mean (μ) : 3 days
- Population std (σ) : 0.6 days
- Sample mean (x bar) : 2.8 days
- Sample size (n) : 45 days
- Significance level (α) : 0.05
- Confidence Interval (CI) : 0.95

Here, since **population std is known** and **sample size>=30**, we can use a Z-test, specifically a two-tailed one-sample z-test.

**5 - Python Implementation (Step-by-Step)**

In [2]:
import numpy as np
from scipy import stats

# Step 1: Define Parameters
population_mean = 3         # Claimed population mean
population_std = 0.6        # Known population standard deviation
sample_mean = 2.8           # Observed sample mean
sample_size = 45            # Sample size
alpha = 0.05                # Significance level

# Step 2: Caculate Standard Error (SE)
standard_error = population_std/np.sqrt(sample_size)
print(f"Standard Error (SE): {standard_error:.4f}")

# Step 3: Compute Z Statistic
z_stat = (sample_mean - population_mean)/standard_error
print(f"Z Statistic: {z_stat:.4f}")

# Step 4: Compute Two-Tailed P-Value
p_value = 2 * (1 - stats.norm.cdf(abs(z_stat)))
print(f"P-Vaue: {p_value:.4f}")

# Step 5: Compute Critical Values for Two-Tailed Test
# Split alpha in half for both sides
z_critical_left = stats.norm.ppf(alpha/2)
z_critical_right = stats.norm.ppf(1-alpha/2)
print(f"Critical Values: Left = {z_critical_left:.4f}, Right = {z_critical_right:.4f}")

# Step 6: Decision Making
print("\n--- Decision Based on P-Value ---")
if p_value < alpha:
    print("Reject H0: The sample mean is significantly different from population mean.")
else:
    print("Fail to Reject H0: No significant difference detected.")

print("\n--- Decision Based on Critical Values ---")
if (z_stat < z_critical_left) or (z_stat > z_critical_right):
    print("Reject H0: The sample mean is significantly different from population mean.")
else:
    print("Fail to Reject H0: No significant difference detected.")

Standard Error (SE): 0.0894
Z Statistic: -2.2361
P-Vaue: 0.0253
Critical Values: Left = -1.9600, Right = 1.9600

--- Decision Based on P-Value ---
Reject H0: The sample mean is significantly different from population mean.

--- Decision Based on Critical Values ---
Reject H0: The sample mean is significantly different from population mean.


**6 - Business Interpretation**
- Z = -2.36 -> The sample mean is **2.36 standard errors below** the population mean.
- p = 0.0253 -> There's only about 2.5% chance of seeing such difference if the true mean were actually 3.

**Conclusion**
Since **p<0.05**, we **Reject H0**, concluding the **average delivery time is signficantly different** form the claimed 3 days.

In business terms: delivery time **has changed**, and the company's claim no logner reflects the real performance.